core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/SampleSheet.java

```diff
@@ -1,24 +1,75 @@
 package uk.ac.bbsrc.tgac.miso.core.util;
 
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
 import java.util.List;
+import java.util.Optional;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 import com.eaglegenomics.simlims.core.User;
 
+import uk.ac.bbsrc.tgac.miso.core.data.IlluminaRun;
 import uk.ac.bbsrc.tgac.miso.core.data.Index;
 import uk.ac.bbsrc.tgac.miso.core.data.Partition;
 import uk.ac.bbsrc.tgac.miso.core.data.Run;
 import uk.ac.bbsrc.tgac.miso.core.data.impl.view.PoolDilution;
 import uk.ac.bbsrc.tgac.miso.core.data.impl.view.PoolableElementView;
 
 public enum SampleSheet {
-  CASAVA_1_7 {
+  BCL2FASTQ("BCL2FASTQ") {
+
+    @Override
+    public boolean allowedFor(Run run) {
+      return run instanceof IlluminaRun;
+    }
+
+    @Override
+    protected Stream<String> getColumns() {
+      return Stream.of("Sample_ID", "Sample_Name", "I7_Index_ID", "index", "I5_Index_ID", "index2");
+    }
+
+    @Override
+    protected String header(Run run) {
+      IlluminaRun r = (IlluminaRun) run;
+      String reads = "";
+      if (r.getSequencingParameters() != null) {
+        String readLength = Integer.toString(r.getSequencingParameters().getReadLength());
+        reads = String.format("[Reads]\n%s\n%s\n\n", readLength,
+            r.getSequencingParameters().isPaired() ? readLength : "");
+      }
+      return String.format("[Header]\nDate,%s\n\n%s[Data]\n", DateTimeFormatter.ISO_DATE.format(ZonedDateTime.now()), reads);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') %n should be used in place of \n to produce the platform-specific line separator. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3457)